### PR TITLE
[rand gen] adjust interface and support persist/re-use aug data

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

**Comment by zekun000** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #11453** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11453?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11453?utm_source=stack-comment).

**Review by zjma — APPROVED** [review]

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 1]
```diff
@@ -213,23 +213,38 @@ impl<
...
+            info!("[RandManager] Already have certified aug data");
+            return certified_data;
+        }
+        let data = self
+            .aug_data_store
+            .get_my_aug_data()
+            .unwrap_or_else(|| D::generate(&self.config));
+        // Add it synchronously to avoid race that it sends to others but panics before it persists locally.
+        self.aug_data_store
+            .add_aug_data(data.clone())
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/rand_manager.rs:227]: If `aug_data_store` already has data, will add fail?

Comment (zekun000) [reply consensus/src/rand/rand_gen/rand_manager.rs:227]: no as long as it’s the same data

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 2]
```diff
@@ -259,7 +274,8 @@ impl<
             )
         );
 
-        let _guard = self.broadcast_aug_data();
+        let certified_data = self.broadcast_aug_data().await;
+        let _guard = self.broadcast_certified_aug_data(certified_data);
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/rand_manager.rs:278]: Move this before spawning `verification_task`?

**Review by danielxiangzl — APPROVED** [review]

File: consensus/src/rand/rand_gen/rand_manager.rs (hunk 2 above)
Comment (zekun000) [reply consensus/src/rand/rand_gen/rand_manager.rs:278]: why? if we don’t spawn verification task we can’t get certified data

Comment (danielxiangzl) [reply consensus/src/rand/rand_gen/rand_manager.rs:278]: Good point. Its just some verification_task needs aug data, but not important

**Pull request merged**